src/renderer/Dataframe.js

```diff
@@ -286,8 +286,8 @@ export default class Dataframe extends DummyDataframe {
         const points = this.decodedGeom.vertices;
         const features = [];
 
-        const WIDTH = this.renderer.gl.canvas.width;
-        const HEIGHT = this.renderer.gl.canvas.height;
+        const WIDTH = this.renderer.gl.canvas.width / window.devicePixelRatio;
```

Comment:
Good catch! 

As we are reusing the same expression 3 times in several places, we could also have:
```js
_getCanvasSize(){
    const WIDTH = this.renderer.gl.canvas.width / window.devicePixelRatio;
    const HEIGHT = this.renderer.gl.canvas.height / window.devicePixelRatio;
    return [WIDTH, HEIGHT]; // or maybe with an object {width, height} 
}
```

And then just call it like `const [WIDTH, HEIGHT] = this._getCanvasSize()` in every place.

Reply:
I'm going to test that fix!

Reply:
Yes... and it looks better. But I'm worried about the performance, I already suffered the cost of allocating new arrays. I don't know if this will be costly though, but it would require to re-check the performance :(

Reply:
I've tested and it works like a charm after the fix

Reply:
> Yes... and it looks better. But I'm worried about the performance, I already suffered the cost of allocating new arrays. I don't know if this will be costly though, but it would require to re-check the performance :(

Oh, I see, I was considering perfomance... 👍